overlord/configstate/config/helpers.go

```diff
@@ -299,6 +317,7 @@ type Conf interface {
 type ConfGetter interface {
 	Get(snapName, key string, result interface{}) error
 	GetMaybe(snapName, key string, result interface{}) error
+	GetNoVirtual(snapName, key string, result interface{}) error
```

Comment:
Is there a use case the caller should call Get or GetNoVirtual? Do we need both (and what about GetMaybeVirtual)? Can we keep it simple and only have Get and GetMaybe?

Reply:
the original PR had a recursion problem, but is a bit unclear whether this new code has it or not. I suppose this should be introduced only once we have a use case.

Reply:
I switched to an alternative approach now that makes this a property of the entire transaction. Please let me know if you feel this is cleaner (it feels cleaner to me). If it does not help I will just revert this commit 747fa0bc2a825ffa7729d82fe038598bd1b7ee60

Reply:
I removed the need for GetNoVirtual in the latest version of the PR entirely.